### DISCUSSION TITLE
2D Simulation

### DISCUSSION THREAD

**Discussion by ocg2347** [raisimTech/raisimLib#288] [Q&A]
Hi to all and thanks to the developers for this simple and powerful simulator!
I'm currently working on a exoskeleton project and using RaiSim for RL training. And for my study, it'd be awesome to constraint the robot to move along the sagittal plane.  (Our robot is designed to move along the sagittal plane only and, by its design, it will not spread its legs in lateral direction.)
So the questions are:
- Is there a built-in functionality for doing this,
- If not, what'd be the best practice to accomplish this?
One obvious method would be updating the corresponding root position and orientation to force it to lie inside the desired plane after every integration step and solve kinematics for the rest of the agent. Any suggestions that might go beyond this?

**Comment by jhwangbo** [comment]
You can make two prismatic joints that connect the world to the base.
